vector/array.go

```diff
@@ -38,3 +38,30 @@ func (a *Array) Serialize(b *zcode.Builder, slot uint32) {
 	}
 	b.EndContainer()
 }
+
+func ArrayRange(val Any, slot uint32) (uint32, uint32, bool) {
+	// Dict / Const do not appear for Arrays?
+	switch val := val.(type) {
+	case *Array:
+		return val.Offsets[slot], val.Offsets[slot+1], val.Nulls.Value(slot)
+	case *View:
+		slot = val.Index[slot]
+		a := val.Any.(*Array)
+		return a.Offsets[slot], a.Offsets[slot+1], a.Nulls.Value(slot)
+	}
+	panic(val)
+}
```

Comment:
@mccanne am I right here in that Dict and Consts are not used for Arrays? Also the function name is kind of weird if you have any better ideas.